plugins/callback/foreman.py

```diff
@@ -281,17 +328,25 @@ def send_reports(self, stats):
                 report['config_report']['status']['pending'] = total['changed']
                 report['config_report']['status']['applied'] = 0
 
-            self._send_data('report', host, report)
+            self._send_data('report', 'foreman', host, report)
 
             self.items[host] = []
 
+    def send_reports(self, stats):
+        if self.report_type == "foreman":
+            self.send_reports_foreman_legacy(stats)
+        elif self.report_type == "proxy":
+            self.send_reports_proxy_host_report(stats)
+        else:
+            self._display.warning(u'Unknown foreman endpoint type: {type}'.format(type=self.report_type))
+
     def append_result(self, result, failed=False):
         name = result._task.get_name()
         host = result._host.get_name()
         value = result._result
         value['failed'] = failed
         value['module'] = result._task.action
```

Comment:
Shell we copy the dictionary? Because if the `_result` method does not return a copy, we are modifying the source dictionary here which is not nice.